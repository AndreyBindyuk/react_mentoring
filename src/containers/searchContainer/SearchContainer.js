import React from 'react';
import "./Search.css";
import MovieStub from '../../MovieStub.json'

class SearchContainer extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            search_by:"title",
            sort_by:"release"
        }
    }

    toggleTitleColor = (event) => {
        if(event.target.id == "btn_genre" && this.state.search_by == "title"){
            this.setState({ search_by: "genre"})
        }
        else if(event.target.id == "btn_title" && this.state.search_by == "genre"){
            this.setState({ search_by: "title"})
        }
    }
    
    toggleRatingColor = (event) => {
        if(event.target.id == "btn_rating" && this.state.sort_by == "release"){
            this.setState({ sort_by: "rating"})
        }
        else if(event.target.id == "btn_release" && this.state.sort_by == "rating"){
            this.setState({ sort_by: "release"})
        }
    }

    render() {
        return (
            <div className="search-container">
                <div className="input-button-container">
                    <input className="input-container" type="input" placeholder="Search..." />
                    <div className="buttons-movies-result-container">
                        <div className="buttons-container">
                            <span className="desc">SEARCH BY</span>
                            <div className="buttons-position">
                                <button id="btn_title" onClick={this.toggleTitleColor} className={this.state.search_by =="title" ? "btn-red" : "btn-gray"}>
                                    TITLE
                                </button>
                                <button id="btn_genre" onClick={this.toggleTitleColor} className={this.state.search_by =="title" ? "btn-gray" : "btn-red"}>
                                    GENRE
                                </button>
                            </div>
                        </div>
                        <div className="buttons-container">
                            <span className="desc">SORT BY</span>
                            <div className="buttons-position">
                                <button id="btn_release" onClick={this.toggleRatingColor} className={this.state.sort_by =="release" ? "btn-red" : "btn-gray"}>
                                    RELEASE DATE
                                </button>
                                <button id="btn_rating" onClick={this.toggleRatingColor} className={this.state.sort_by =="release" ? "btn-gray" : "btn-red"}>
                                RATING
                            </button>
                            </div>
                        </div>
                        <div className="movies-search-result">
                            <div className="result-count"><span>{MovieStub.length}</span>results</div>
                            <button className="btn-search">SEARCH</button>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

export default SearchContainer;