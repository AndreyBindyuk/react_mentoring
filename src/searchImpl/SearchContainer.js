import React from 'react';
import "../css/Search.css";
import mainLogo from '../movie_pic.png';
import MovieStub from '../movieItemImpl/MovieStub.json'

class SearchContainer extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            btn_title_genre: true,
            btn_release_rating: true
        }
    }

    changeBtnColor(params) {
        if ((params =="btn_title" || params =="btn_genre") && document.getElementById(params).style.color != "red"){
            this.setState({ btn_title_genre: !this.state.btn_title_genre })
        } else if((params =="btn_release" || params =="btn_rating") && document.getElementById(params).style.color != "red"){
            this.setState({ btn_release_rating: !this.state.btn_release_rating })
        }
    }

    render() {
        let red_btn = this.state.btn_title_genre ? "red" : "";
        let gray_btn = this.state.btn_title_genre ? "" : "red";
        let r_red_btn = this.state.btn_release_rating ? "red" : "";
        let r_gray_btn = this.state.btn_release_rating ? "" : "red";

        return (
            <div className="search-container">
                <img className="logo-container" src={mainLogo} />
                <div className="input-button-container">
                    <input className="input-container" type="input" />
                    <div className="buttons-container">
                        <div className="title-genre-button-container">
                            <button id="btn_title" onClick={this.changeBtnColor.bind(this, "btn_title")} style={{ color: red_btn }}>
                                <span>TITLE</span>
                            </button>
                            <button id="btn_genre" onClick={this.changeBtnColor.bind(this, "btn_genre")} style={{ color: gray_btn }}>
                                <span>GENRE</span>
                            </button>
                        </div>
                        <button><span>SEARCH</span></button>
                    </div>
                    <div className="movies-search">
                        <h5>{MovieStub.length} movies found</h5>
                        <div className="release-rating">
                            <button id="btn_release" className="btn" style={{ color: r_red_btn }} onClick={this.changeBtnColor.bind(this, "btn_release")}>release date</button>
                            <button id="btn_rating" className="btn" style={{ color: r_gray_btn }} onClick={this.changeBtnColor.bind(this, "btn_rating")}>rating</button>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

export default SearchContainer;