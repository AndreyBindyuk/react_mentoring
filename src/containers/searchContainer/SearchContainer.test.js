import React from "react";
import { shallow, mount } from "enzyme";
import SearchContainer from "./SearchContainer";

describe("<SearchContainer />", () => {
  it("should render components correctly", () => {
    const component = shallow(<SearchContainer />);
    expect(component).toMatchSnapshot();
  });

  it("should be call toggleTitleColor", () => {
    const component = shallow(<SearchContainer />);
    component.instance().toggleTitleColor = jest.fn();
    component.instance().forceUpdate();
    component.update();
    const mockedEvent = { target: {} };
    component.find("button#btn_genre").simulate("click", mockedEvent);
    expect(component.instance().toggleTitleColor).toHaveBeenCalled();
  });

  it("search_by /genre/ parameter shoud be changed", () => {
    const component = mount(<SearchContainer />);
    component.instance().toggleTitleColor = jest.fn();
    const mockedEvent = { target: { id: "btn_genre" } };
    component.find("button#btn_genre").simulate("click", mockedEvent);
    expect(component.state().search_by).toEqual("genre");
  });

  it("search_by /title/ parameter shoud be changed", () => {
    const component = mount(<SearchContainer />);
    component.setState({ search_by: "genre" });
    const mockedEvent = { target: { id: "btn_title" } };
    component.find("button#btn_title").simulate("click", mockedEvent);
    expect(component.state().search_by).toEqual("title");
  });

  it("sort_by /rating/ parameter shoud be changed", () => {
    const component = mount(<SearchContainer />);
    component.instance().toggleRatingColor = jest.fn();
    const mockedEvent = { target: { id: "btn_rating" } };
    component.find("button#btn_rating").simulate("click", mockedEvent);
    expect(component.state().sort_by).toEqual("rating");
  });

  it("sort_by /release/ parameter shoud be changed", () => {
    const component = mount(<SearchContainer />);
    component.setState({ sort_by: "rating" });
    const mockedEvent = { target: { id: "btn_release" } };
    component.find("button#btn_release").simulate("click", mockedEvent);
    expect(component.state().sort_by).toEqual("release");
  });
});
