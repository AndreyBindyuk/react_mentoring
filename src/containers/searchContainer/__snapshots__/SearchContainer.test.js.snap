// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchContainer /> should render components correctly 1`] = `ShallowWrapper {}`;
